backend/src/main/java/com/redhat/cloud/notifications/db/EventResources.java

```diff
@@ -158,25 +190,69 @@ private static void setQueryParams(Mutiny.Query<?> query, String accountId, Set<
             query.setParameter("endDate", endDate);
         }
         if (endpointTypes != null && !endpointTypes.isEmpty()) {
-            query.setParameter("endpointTypes", endpointTypes);
+            query.setParameter("endpointTypes", endpointTypes.stream().map(EndpointType::ordinal).collect(Collectors.toSet()));
         }
         if (invocationResults != null && !invocationResults.isEmpty()) {
             query.setParameter("invocationResults", invocationResults);
         }
     }
 
-    private static String getSortField(String field) {
+    private static String getSortField(String field, boolean nativeQuery) {
         switch (field) {
             case "bundle":
-                return "b.displayName";
+                return nativeQuery ? "b.display_name" : "b.displayName";
             case "application":
-                return "a.displayName";
+                return nativeQuery ? "a.display_name" : "a.displayName";
             case "event":
-                return "et.displayName";
+                return nativeQuery ? "et.display_name" : "et.displayName";
             case "created":
                 return "e.created";
             default:
                 throw new IllegalArgumentException("Unknown sort field: " + field);
         }
     }
+
+    private static List<UUID> parseEventIds(List<Object[]> records) {
```

Comment:
Probably worth adding a note that records[0] has the stuff we need and records[1] has the same count.
Maybe we could even use the stream API to only pass the events and/or to create the result?

Reply:
Good point, I added a comment.

Since we're getting rid of the reactiveness soon, I'd rather keep the number of streams as low as possible 😃 They come with many drawbacks (performances, stacktrace...).